File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/service/RestaurantService.java
package service;

import models.Restaurant;

import java.util.HashMap;
import java.util.Map;

public class RestaurantService {
    private Map<String, Restaurant> restaurants = new HashMap<>();

    public void addRestaurant(Restaurant restaurant) {
        restaurants.put(restaurant.getId(), restaurant);
    }

    public Restaurant getRestaurantById(String id) {
        return restaurants.get(id);
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/service/src/RestaurantService.cpp
#include "../include/RestaurantService.h"

void RestaurantService::addRestaurant(const std::shared_ptr<Restaurant>& restaurant) {
    restaurants[restaurant->getId()] = restaurant;
}

const std::shared_ptr<const Restaurant> RestaurantService::getRestaurantById(const std::string& id) const {
    auto it = restaurants.find(id);
    if (it != restaurants.end()) {
        return it->second;
    }
    return nullptr;
}